### ISSUE TITLE
Construction:WindowEquivalentLayer using WindowMaterial:SimpleGlazingSystem as a material fatals out silently

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#7048]
Issue overview
--------------
User file has a `Construction:WindowEquivalentLayer` which references a `WindowMaterial:SimpleGlazingSystem` as a material layer. The run fatals out with the following err output:
```
Program Version,EnergyPlus, Version 9.0.1-bb7ca4f0da, YMD=2018.11.06 16:36,
   ************* Warning:  Node connection errors not checked - most system input has not been read (see previous warning).
   ************* Fatal error -- final processing.  Program exited before simulations began.  See previous error messages.
   ************* EnergyPlus Warmup Error Summary. During Warmup: 0 Warning; 0 Severe Errors.
   ************* EnergyPlus Sizing Error Summary. During Sizing: 0 Warning; 0 Severe Errors.
   ************* EnergyPlus Terminated--Fatal Error Detected. 0 Warning; 0 Severe Errors; Elapsed Time=00hr 00min  0.08sec
```

More info would be useful here to inform the user that SimpleGlazingSystem is not valid here. Use `Construction`.

### Details
Some additional details for this issue (if relevant):
 - Win64
 - Version of EnergyPlus v9.0.1
 - Helpdesk ticket number 13102

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by Nigusse** [comment]
@mjwitte I added a check for valid Layers Type for Construction:WindowEquivalentLayer object. But when I run the defect in develop (V9.2), JSON Validator catches the invalid layer  and throws the following error before it gets to CONSTRUCTION object getinput function:

   ** Severe  ** <root>[Construction] - Object contains properties that could not be validated using 'properties' or 'additionalProperties' constraints
   ** Severe  ** <root>[Construction] - Object name is required and cannot be blank or whitespace
   **  Fatal  ** Errors occurred on processing input file. Preceding condition(s) cause termination.

I tested the defect file using V9.0 and I duplicated the error message that you posted. I am wondering why JSON did not catch it before in V9.1. Is it worth fixing the code to check for layers if JSON already does it?

**Comment by mjwitte** [comment]
@Nigusse Something has changed in the input processing that is now validating against the object-list properties in the schema. I'm not sure that we're ready for that, but that's a different discussion. Need to figure out when that was added. Beyond that, however, I don't find the above error message very useful to identify what exactly is wrong. When this error fires, is it possible to add more context?

**Comment by Nigusse** [comment]
@MJW I am not sure if we can easily add more context without expanding JSON code base. [Here](https://github.com/NREL/EnergyPlus/blob/develop/src/EnergyPlus/InputProcessing/InputValidation.cc#L102-L105) is the code where the error message is generated.

**Comment by Nigusse** [comment]
@WJM I agree that the error message is not helpful in identifying the offending object. It is generic error message. It is very difficult to find which construction object has the problem if you have multiple objects. I was able to identify the problem from the error message you posted. This will be challenging especially for novice users.

**Comment by mjwitte** [comment]
Hmmm, that error message was added as part of [changes to enforce required fields](https://github.com/NREL/EnergyPlus/commit/6b62aefb61d61a839d28d68431464261a547ae2f) as part of https://github.com/NREL/EnergyPlus/pull/7114. This shouldn't be tripping here, unless you've got a runperiod with a blank name?

**Comment by Nigusse** [comment]
I used the defect file in github. I just checked it, the Runperiod object name field is not blank.

**Comment by Nigusse** [comment]
@mjwitte Hmmm. I found the problem. The validator was not happy with a character/symbol in the object name. The offender is "solaio pavimento(1°piano)/2°piano," CONSTRUCTION. When I removed the "degree" symbol, the validator runs fine.  OK will proceed fixing the main issue.

**Comment by mjwitte** [comment]
OK- but does that mean we need a different fix to allow those characters? @Myoldmopar @mbadams5

**Comment by Nigusse** [comment]
I think we do need a fix, or we need some where in the documentation that tells what characters / symbols are not allowed in object names.

**Comment by mbadams5** [comment]
I'll have to look at the defect file because if the characters are UTF-8, it should just work. If they are not UTF-8 then it should still not be a blank name.

**Comment by mjwitte** [comment]
UTF-8 question forked into new issue #7388

**Comment by Myoldmopar** [comment]
Closed via #7357
